hazelcast/src/test/resources/hazelcast-fullconfig-without-network.xml

```diff
@@ -935,12 +935,6 @@
         <statistics-enabled>false</statistics-enabled>
     </pn-counter>
 
-    <split-brain-protection enabled="true" name="member-count-split-brain-protection">
-        <protect-on>READ_WRITE</protect-on>
-        <minimum-cluster-size>3</minimum-cluster-size>
-        <member-count-split-brain-protection/>
-    </split-brain-protection>
-
```

Comment:
Why do we remove this? The rest of the changes I guess is about using non-defaults.

Reply:
This `member-count-split-brain-protection` isn't parsed anywhere, so I didn't schematize it and removed from the  `hazelcast-fullconfig-without-network.yaml` file. Then I also had to remove it from this file as well, because otherwise the `XmlYamlConfigBuilderEqualsTest#testFullConfigNormalNetwork()` and `#testFullConfigAdvancedNetwork()` did not pass: the `xmlConfigFromXml` (generated from the `xmlConfig`) included this additional `split-brain-protection` element, while the `xmlConfigFromYaml` (generated from `yamlConfig`) did not.

For clarity, `xmlConfigFromXml` prints this for this specific `split-brain-protection`:

```xml
    <split-brain-protection name="member-count-split-brain-protection" enabled="true">
          <minimum-cluster-size>3</minimum-cluster-size>
          <protect-on>READ_WRITE</protect-on>
     </split-brain-protection>
```
So the problematic `<member-count-split-brain-protection/>` doesn't appear here as well. Also, 

I found that the `SplitBrainProtectionFunction` has indeed a `MemberCountSplitBrainProtectionFunction` implementation, but there are no usages of it.

So in short, it looks to be a half-done or unsupported feature.